File: atlas/examples/typescript/packages/application/src/catalog/CatalogRepository.ts
import type { CatalogItem } from '@atlas-example/domain';

/**
 * Persists and retrieves catalog items for application use cases.
 */
export interface CatalogRepository {
  /**
   * Stores one item by stable identity.
   *
   * @param item - Domain item that replaces any item with the same identifier.
   */
  save(item: CatalogItem): void;

  /**
   * Lists all stored items in repository-defined deterministic order.
   *
   * @returns Immutable item view.
   */
  list(): readonly CatalogItem[];
}

File: atlas/examples/kotlin/application/src/main/kotlin/dev/atlas/example/application/catalog/CatalogRepository.kt
package dev.atlas.example.application.catalog

import dev.atlas.example.domain.catalog.CatalogItem

/**
 * Persists and retrieves catalog items for application use cases.
 */
interface CatalogRepository {
    /**
     * Stores one item by stable identity.
     *
     * @param item Domain item that replaces any item with the same identifier.
     */
    fun save(item: CatalogItem)

    /**
     * Lists all stored items in repository-defined deterministic order.
     *
     * @return Immutable item view.
     */
    fun list(): List<CatalogItem>
}
